src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -134,6 +161,143 @@ public Command stowTurret() {
     return new InstantCommand(() -> setTurretPos(Math.toRadians(0))).withName("StowTurretCommand");
   }
 
+  public Command zeroTurretCommand() {
+    return Commands.runOnce(() -> this.zeroTurret(), this).withName("ZeroTurretCommand");
+  }
+
+  public double calculateTurretPosition() {
+    double encoderPosition =
+        lowGearCANcoder.getPosition().getValueAsDouble()
+            / Constants.TurretConstants.LOW_GEAR_CAN_CODER_RATIO;
+    return encoderPosition + Constants.TurretConstants.ENCODER_ZERO_OFFSET_FROM_TURRET_ZERO_REV;
+  }
+
+  /**
+   * A command to set the current turret position as encoder zero.
+   *
+   * @return a command that sets the current position as encoder zero
+   */
+  public Command zeroZeroing() {
+    return Commands.runOnce(
+        () -> {
+          MagnetSensorConfigs cfg = new MagnetSensorConfigs();
+          cfg.withAbsoluteSensorRange(AbsoluteSensorRangeValue.Unsigned_0To1);
+          cfg.withMagnetOffset(0);
+          CANcoderConfigurator lowGearCANcoderConfigurator = lowGearCANcoder.getConfigurator();
+          CANcoderConfigurator highGearCANcoderConfigurator = highGearCANcoder.getConfigurator();
+          lowGearCANcoderConfigurator.apply(cfg);
+          highGearCANcoderConfigurator.apply(cfg);
+
+          final double trueZeroLowGearOffset =
+              lowGearCANcoder.getAbsolutePosition().getValueAsDouble();
+          final double trueZeroHighGearOffset =
+              highGearCANcoder.getAbsolutePosition().getValueAsDouble();
+
+          final double lowGearOffset =
+              trueZeroLowGearOffset
+                  - TurretConstants.LOW_GEAR_CAN_CODER_RATIO
+                      * TurretConstants.ENCODER_ZERO_OFFSET_FROM_TURRET_ZERO_REV;
+          final double highGearOffset =
+              trueZeroHighGearOffset
+                  - TurretConstants.HIGH_GEAR_CAN_CODER_RATIO
+                      * TurretConstants.ENCODER_ZERO_OFFSET_FROM_TURRET_ZERO_REV;
```

Comment:
Is this zeroing from facing forward position? Add comments about what position the turret should be in when this Command should be run

Reply:
I haven't made a decision regarding that, it would be determined by the encoder zero to true zero offset. So the correct orientation is still a open question.